### DISCUSSION TITLE
Pygimli Inversion

### DISCUSSION THREAD

**Discussion by makeabhishek** [gimli-org/pyGIMLi#624] [Q&A] [answered]
Hi Pygimli team, 

I have some fundamental questions regarding the inversion framework.
I was reading the article "Rücker, Carsten, Thomas Günther, and Florian M. Wagner. "pyGIMLi: An open-source library for modelling and inversion in geophysics." Computers & Geosciences 109 (2017): 106-123."

It is mentioned that "The default inversion framework is based on the generalized Gauss-Newton method" and further that the application of the Gauss-Newton scheme on minimizing the objective function` (\phi)` yields the model update in the iteration which is solved using a conjugate-gradient least-squares solver. Aren't Gauss-Newton and conjugate gradient different methods?

<img src="https://github.com/gimli-org/gimli/assets/47937684/75670a05-69a2-4a95-b87f-750603930be3" width="40%" height="40%"> \
<img src="https://github.com/gimli-org/gimli/assets/47937684/54708d73-c5da-42c1-8f3a-c212c8cf0337" width="40%" height="40%">

The system of above equations has to be solved in every iteration step.
How is the conjugate-gradient method incorporated with the Gauss-Newton method?

Furthermore, how is the smoothness `(\phi_m) `model defined in Pygimli, used in the objective function? Is it a kind of diagonal matrix?

How the implemented inverse optimisation method is different from conventional Gauss-newton method?

**Comment by halbmy** [comment]
Gauss-Newton is a minimization method, i.e. a way to iteratively find a model update using the Jacobian matrix (2nd derivative), whereas there are also gradient-based minimization schemes like NLCG (nonlinear conjugate gradients).

So solve the inverse subproblem, we use a conjugate-gradient least-squares solver.

The smoothness matrix is a sparse matrix defining  1st or 2nd (or 0th or mixes of them) order derivatives.

For details on all this please have a look at Günther et al. (2006).

Günther, T., Rücker, C. & Spitzer, K. (2006): Three-dimensional modeling and inversion of dc resistivity data in-
corporating topography – II: Inversion. Geophys. J. Int. 166, 506-517,  <http://doi.org/10.1111/j.1365-246X.2006.03011.x>.

**Reply by makeabhishek** [reply]
Thanks for the explanation. Yes I was reading the referred article. you mentioned, " Gauss-Newton is a minimization method, i.e. a way to iteratively find a model update using the Jacobian matrix (2nd derivative)". Isn't Jacobian define the 1st derivation and Heasian define 2nd derivative? 
However, when I read "Rücker, Carsten, Thomas Günther, and Florian M. Wagner.  (2017): pyGIMLi: An open-source library for modelling and inversion in geophysics." Computers & Geosciences, I get confused. Because it mentioned that,

 inversion uses Gauss-Newton method for minimisation and then solved using a conjugate-gradient least-squares solver, as shown in the below image:

<img src="https://github.com/gimli-org/gimli/assets/47937684/dd59b42c-53b4-4fc8-a3ea-1ece9428270b" width="40%" height="40%">

**Reply by halbmy** [reply]
There is no contradiction. The Jacobian `J` is the derivative of the forward model with respect to the model parameters and the Gauss-Newton method approximates the Hessian (2nd derivative of the objective function) by `J.T J`. It is written in many text books and papers, but maybe it is a good idea to write it all down in a tutorial, where the individual components are introduced and shown.

**Reply by makeabhishek** [reply]
Is the shortest ray path method used in refraction manages same as `Dijkstra's algorithm`?

**Reply by halbmy** [reply, marked as answer]
Yes, with secondary notes as explained in Girou & Laroche

**Reply by halbmy** [reply]
I guess the original question has been solved and the issue van solved /close

**Comment by makeabhishek** [comment]
I just want to check the final thing. What is the abbreviation for the name `CGLSCDWWtrans`, as mentioned in #712? I know `CGLS` refers conjugate gradient least squares, but didn't understand `DWWtrans` meaning.

I read your PhD thesis but found it difficult to find this. May I request that you refer me to the relevant section?

**Reply by halbmy** [reply]
In my PhD thesis, Algorithm 6 is about CGLSCDP. WW means weighting the regularization from two sides (boundary and model control) and trans the data and model transformation functions.

**Comment by makeabhishek** [comment]
Just want to add here for reference. Some information about constraint/smoothness matrix was discussed in #688.
